### DISCUSSION TITLE
When rendering the score do all the notes on in sequence have a progressive id?

### DISCUSSION THREAD

**Discussion by gerardoscoppetta** [opensheetmusicdisplay/opensheetmusicdisplay#1161] [Q&A]
I would like to be able to synchronize the midi notes of a web midi sequencer with the cursor highlighting the notes. I would just need to have the chronological sequence of all the notes without counting the rests. Is there a way to derive this chronological sequence of the notes? Thanks

**Comment by infojunkie** [comment]
I did something similar here https://blog.karimratib.me/demos/musicxml/ - code is linked from the demo. Basically, I needed to calculate the cursor's measure index and voice index to position the cursor manually.

**Comment by gerardoscoppetta** [comment]
Thanks. I try immediately.

**Comment by sschmidTU** [comment]
You could also just call `osmd.cursor.next()` in a loop to go through all the notes of the piece. When you've reached the end, `osmd.cursor.Iterator.EndReached` will be true. See `osmd.cursor.NotesUnderCursor()` - there is also an isRest() function on notes.

**Reply by gerardoscoppetta** [reply]
Is it not possible to jump from one area to another? For example, if I set a loop on the midi file from measure 4 to measure 9 of a song and I want to repeat the musical phrase, I would need to be able to direct the cursor in that area. I saw this "array [0] ['NoteToGraphicalNoteObjectId']" which returns the progressive id of the notes but I don't see the possibility to send the cursor where I want. Is there such a possibility?
Thanks.

**Reply by sschmidTU** [reply]
Yes, that's possible, you didn't ask about that before. Either call `cursor.next()` you're at the right spot (after calling `cursor.reset()`), or if you want it a little more elegant, create a MusicPartManagerIterator with the startTimestamp as second argument (first is osmd.Sheet), and set that as `curser.iterator`. Then call cursor.update().

**Reply by gerardoscoppetta** [reply]
Perfect thanks. I try.

**Reply by gerardoscoppetta** [reply]
I succeeded with the first method he suggested to me. After a certain number of notes I send it back with cursor.reset (). Is there a quick way to send it anywhere in the score? for example in bar 16 on the first movement. Thanks so much.

**Reply by sschmidTU** [reply]
Basically the same as i wrote initially (with the loop) - just check for `osmd.cursor.Iterator.CurrentMeasure` instead of EndReached

**Reply by sschmidTU** [reply]
By the way, all of this is much easier with the OSMD Audio Player, where you can just set the cursor to a timestamp directly (the measure's timestamp). It's currently in early access for Github sponsors.

**Reply by gerardoscoppetta** [reply]
Ok thanks again. Now I try.
